packages/jest-config/src/index.ts

```diff
@@ -267,6 +229,91 @@ This usually means that your ${chalk.bold(
   }
 };
 
+export interface ReadJestConfigOptions {
+  /**
+   * The package root or deserialized config (default is cwd)
+   */
+  packageRootOrConfig?: string | Config.InitialOptions;
+  /**
+   * When the `packageRootOrConfig` contains config, this parameter should
+   * contain the dirname of the parent config
+   */
+  parentConfigDirname?: null | string;
+  /**
+   * Indicates whether or not to read the specified config file from disk.
+   * When true, jest will read try to read config from the current working directory.
+   * (default is false)
+   */
+  readFromCwd?: boolean;
+  /**
+   * Indicates whether or not to ignore the error of jest finding multiple config files.
+   * (default is false)
+   */
+  skipMultipleConfigError?: boolean;
+}
+
+/**
+ * Reads the jest config, without validating them or filling it out with defaults.
+ * @param config The path to the file or serialized config.
+ * @param param1 Additional options
+ * @returns The raw initial config (not validated)
+ */
+export async function readInitialOptions(
+  config?: string,
+  {
+    packageRootOrConfig = process.cwd(),
+    parentConfigDirname = null,
+    readFromCwd = false,
+    skipMultipleConfigError = false,
+  }: ReadJestConfigOptions = {},
+): Promise<{config: Config.InitialOptions; configPath: string | null}> {
+  if (typeof packageRootOrConfig !== 'string') {
+    if (parentConfigDirname) {
+      const rawOptions = packageRootOrConfig;
+      rawOptions.rootDir = rawOptions.rootDir
+        ? replaceRootDirInPath(parentConfigDirname, rawOptions.rootDir)
+        : parentConfigDirname;
+      return {config: rawOptions, configPath: null};
+    } else {
+      throw new Error(
+        'Jest: Cannot use configuration as an object without a file path.',
+      );
+    }
+  } else if (isJSONString(config)) {
```

Comment:
can we avoid the `else-if` and just have `if`? From what I can see, we always `return` or `throw`, so the `else` part does nothing

Reply:
Sure! Fixed with the latest push.